docs/components/UserContext.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { createContext, useEffect, useState } from "react";
+
+type User = { address: string };
+
+type Context = {
+  user: User | null;
+  hasGlow: boolean;
+
+  signIn: () => Promise<void>;
+  signOut: () => Promise<void>;
+};
+
+export const UserContext = createContext<Context | null>(null);
+
+export const UserProvider = ({ children }: { children: React.ReactNode }) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  const [hasGlow, setHasGlow] = useState(true);
+  useEffect(() => {
+    let timeout: number | undefined = undefined;
+
+    if (window.glow) {
+      setHasGlow(true);
+    } else {
+      setHasGlow(false);
+
+      timeout = window.setTimeout(() => {
+        if (window.glow) {
+          setHasGlow(true);
+        }
+      }, 2000);
+    }
+
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, []);
```

Comment:
This actually has some minor issues, if we want to use `setTimeout` we can have a special hook. Much more info in this blog post: https://overreacted.io/making-setinterval-declarative-with-react-hooks/